File: apps/api/app/schemas/posts.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SourceVideoOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    external_video_id: str
    permalink: str
    caption: str | None
    posted_at: datetime | None
    duration_seconds: int | None
    stats: dict | None


class ContentAnalysisOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    primary_topic: str
    secondary_topics: list[str]
    content_format: str
    presentation_style: list[str]
    hook_text: str
    hook_type: str
    hook_mechanism: str
    tone: list[str]
    story_structure: list[str]
    audience_promise: str
    emotional_angle: str | None
    cta_pattern: str | None
    editing_intensity: str
    estimated_pacing: str
    personal_story_level: float
    educational_level: float
    visual_analysis_available: bool
    transcript_available: bool
    confidence: float


class OwnPostOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    processing_status: str
    processing_error: str | None
    source_video: SourceVideoOut
    content_analysis: ContentAnalysisOut | None
    created_at: datetime

File: apps/web/types/posts.ts
export interface SourceVideoOut {
  external_video_id: string;
  permalink: string;
  caption: string | null;
  posted_at: string | null;
  duration_seconds: number | null;
  stats: Record<string, unknown> | null;
}

export interface ContentAnalysisOut {
  primary_topic: string;
  secondary_topics: string[];
  content_format: string;
  presentation_style: string[];
  hook_text: string;
  hook_type: string;
  hook_mechanism: string;
  tone: string[];
  story_structure: string[];
  audience_promise: string;
  emotional_angle: string | null;
  cta_pattern: string | null;
  editing_intensity: "low" | "medium" | "high";
  estimated_pacing: "slow" | "medium" | "fast";
  personal_story_level: number;
  educational_level: number;
  visual_analysis_available: boolean;
  transcript_available: boolean;
  confidence: number;
}

export type ProcessingStatus = "pending" | "processing" | "completed" | "failed";

export interface OwnPostOut {
  id: string;
  processing_status: ProcessingStatus;
  processing_error: string | null;
  source_video: SourceVideoOut;
  content_analysis: ContentAnalysisOut | null;
  created_at: string;
}
